### PR TITLE
add error dialogs for uncaught network errors

### PR DESCRIPTION
**Pull request by BelfordZ**
@whilei addresses #614 
errordialog

### CODE REVIEW

File: src/store/network/index.js [hunk 1]
```diff
@@ -2,4 +2,24 @@ import * as selectors from './selectors';
...
-export default { actions, selectors, reducer };
+import screen from '../wallet/screen';
+
+const wrapActionWithErrorHandling = (method) => {
+  return (...props) => {
+    return (dispatch, getState, api) => {
+      const returnVal = method(...props, dispatch, getState, api);
+      if (returnVal.catch) {
+        returnVal.catch((e) => dispatch(screen.actions.showError(e)));
+      }
```
Comment (shanejonas) [inline src/store/network/index.js]: you forgot to return `returnVal` here

File: src/store/network/networkActions.js [hunk 2]
```diff
@@ -6,100 +6,99 @@ import createLogger from '../../utils/logger';
 import ActionTypes from './actionTypes';
 import history from '../wallet/history';
 
+import network from './';
```
Comment (shanejonas) [inline src/store/network/networkActions.js]: this would cause a circular dependency?

Comment (BelfordZ) [reply src/store/network/networkActions.js]: yes,  but webpack doesnt seem to care...

Comment (BelfordZ) [reply src/store/network/networkActions.js]: really hard to make this work without this.

Comment (shanejonas) [reply src/store/network/networkActions.js]: you have these actions in this file?

Comment (BelfordZ) [reply src/store/network/networkActions.js]: yes, but they are wrapped in the index.js 

I could wrap them within this file, and call the wrapped variant instead I suppose

File: src/store/wallet/screen/screenReducers.js [hunk 3]
```diff
@@ -21,9 +21,7 @@ function onOpen(state, action) {
 
 function onError(state, action) {
   if (action.type === 'SCREEN/ERROR') {
-    return state
-      .set('error', action.message)
-      .set('dialog', null).set('dialogItem', null);
```
Comment (BelfordZ) [inline src/store/wallet/screen/screenReducers.js]: Not sure why it was setting this to null. Maybe this was another bug fix? Should I put this back?

**Pull request merged**